Notification Entities + Swift Mk 1: Managed Object

jleandroperez opened the pull request:
Re-submitting PR #5831 (already validated + merged).

@kwonye mind taking a peek at this one?. 
Travis issues should be fixed. Long story short:

- I've managed to reproduce the failure locally
- It's not a bug introduced by the PR itself, but (somehow) it gets exposed!
- Tracked down to an unhandled NSException being thrown, while running the unit tests that don't even access Core Data
- Apparently related to the app being launched simultaneously while running (random) tests

I've added a try/catch block in the `TestContextHelper`, so that the exception gets logged, still, doesn't crash Travis.

cc @astralbodies  @sendhil

kwonye commented:
Yea, this one is tricky because it _looks_ so unrelated, yet it so consistently passes/fails based on this PR.

@sendhil I'm not sure if cf3afa5 is deemed ok, do you mind giving your thoughts on this? It looks like it could just be masking the problem.

kwonye commented on WordPress/WordPressTest/WPAvatarSourceTest.m:
```diff
@@ -48,7 +48,7 @@ - (void)testGravatarCaches
         XCTAssertEqual(image.size.width, 48.f, @"avatar should be resized");
     }];
-    [self waitForExpectationsWithTimeout:5.0 handler:nil];
+    [self waitForExpectationsWithTimeout:30.0 handler:nil];
```
30 is a bit long. Can we go to 10 max?

jleandroperez replied:
Actually, this was totally unrelated. Up in that class we're mocking the response, so... i'll revert this commit, thanks!

jleandroperez commented:
cc @astralbodies

jleandroperez commented:
@kwonye @astralbodies @sendhil Fixed, Travis is all green:

- Updated Unit Tests's **tearDown** method:
 - Added **mainContext.reset()** calls here and there
 - Explicitly removing the Core Data overrides
- Nuked Travis's Cache

Mind taking a peek?

Thanks gentlemen!!!

sendhil commented:
I'm fine with the fix.

jleandroperez commented:
Thanks Sendhil!
